network/serde/jsonrpc.go

```diff
@@ -0,0 +1,105 @@
+package serde
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+)
+
+type RequestMethod string
+
+const (
+	DirectFundRequestMethod    RequestMethod = "direct_fund"
+	DirectDefundRequestMethod  RequestMethod = "direct_defund"
+	VirtualFundRequestMethod   RequestMethod = "virtual_fund"
+	VirtualDefundRequestMethod RequestMethod = "virtual_defund"
+	PayRequestMethod           RequestMethod = "pay"
+)
+
+const JsonRpcVersion = "2.0"
+
+type MessageType int8
+
+const (
+	TypeRequest  MessageType = 1
+	TypeResponse MessageType = 2
+	TypeError    MessageType = 3
+)
+
+// JsonRpcMessage is union of jsonrpc request, response, and error
```

Comment:
Nit: I think it is an _intersection_, not a union (at least that is the terminology from typescript...)

Reply:
Do you think this comment still applies if we disregard Typescript terminology?

Reply:
I'm not really sure 🤷 . Anyway i think it's fine to leave as is, or we could say `JsonRpcMessage has all of the properties of jsonrpc request, response, and error` or something?